### PR TITLE
Use relative paths for replace directives

### PR DESCRIPTION
**Pull request by drewolson**
See title.

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -164,13 +164,12 @@ writeSupportFiles baseOutpath = do
...
     goModSourceExists <- doesFileExist goModSource
     when (not goModSourceExists) $ do
-      currentDir <- getCurrentDirectory
       project <- projectEnv
       let project' = T.unpack project
           outputdir' = tail $ T.unpack outputdir
           modText' = T.replace "$PROJECT" project $ T.decodeUtf8 modText
-          replaceLoader = project' </> ffiLoader' <> "=" <> currentDir </> subdir
-          replaceOutput = project' </> modPrefix' <> "=" <> currentDir </> outputdir'
+          replaceLoader = project' </> ffiLoader' <> "=." </> subdir
```
Comment (andyarvanitis) [inline app/Main.hs:171]: I think for proper portability and platform support, we shouldn't use a dot (`"."`) to denote relative path, but instead use something like `makeRelativeToCurrentDirectory`. I've been sloppy in some places with explicit `/` characters in paths too, so I should fix those.

Comment (drewolson) [reply app/Main.hs:171]: Interestingly, if I update the code to use `makeRelativeToCurrentDirectory`, the replace statements look like this (they have no prefix):

```
replace project.localhost/purescript-native/ffi-loader => purescript-native

replace project.localhost/purescript-native/output => output
```

Now, `go mod` complains that relative paths _must_ start with `.`. In this case, I think we may want the explicit dot in the code, even though it seems wrong.

Thoughts? I'm not a haskell expert nor a `go mod` expert.

Comment (andyarvanitis) [reply app/Main.hs:171]: Yeah, I'm not sure either. I can help take a look at it after the holidays.

**Comment by drewolson** [comment]
Sure, no rush!

**Comment by andyarvanitis** [comment]
I dug into it a bit more, and I think the explicit dot should be ok. It works from the standard Windows command prompt, and I'm guessing the go tools would probably (maybe?) handle it on any other platform anyway. Thanks!

**Pull request merged**